### PR TITLE
cloud-config yaml users/groups reworked for more creation consistency

### PR DESCRIPTION
**Pull request by luthermonson**
The service user we made when merging cloud config yaml would fail in certain situations surrounding the group of the user. defaults for instance of auto-creating a group with the same name of the user would fail to create any user at all if the group already existed. This is just default behavior from useradd so for consistency I've added the groups entry AND the users entry and forced all group creation logic to be off specifying which exact group we will add the new user too.

### CODE REVIEW

File: drivers/vmwarevsphere/cloudinit.go [hunk 1]
```diff
@@ -227,31 +227,50 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
...
 		users := make([]interface{}, 1)
 		users[0] = newUser
 		cf["users"] = users
 	}
 
+	if val, ok := cf["groups"]; ok {
+		g := val.([]interface{})
+		var exists = false
+		for _, v := range g {
+			if v.(string) == group {
```
Comment (rmweir) [inline drivers/vmwarevsphere/cloudinit.go]: may want to safely convert v, with something like:
`if something, _ := v.(string); something == group {`

Comment (luthermonson) [reply drivers/vmwarevsphere/cloudinit.go]: great suggestion

File: drivers/vmwarevsphere/cloudinit.go [hunk 2]
```diff
@@ -227,31 +227,50 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
...
 	}
 
 	newUser := map[interface{}]interface{}{
-		"name":        user,
-		"sudo":        "ALL=(ALL) NOPASSWD:ALL",
-		"lock_passwd": "true",
+		"name":          user,
+		"sudo":          "ALL=(ALL) NOPASSWD:ALL",
+		"lock_passwd":   true,
+		"groups":        group,
```
Comment (daxmc99) [inline drivers/vmwarevsphere/cloudinit.go]: What spec is this implementing? It might be worth a comment since this is an `interface{}`

**Comment by daxmc99** [comment]
What you wrote in the PR comment should also be in commit I think. That adds alot of context

File: drivers/vmwarevsphere/cloudinit.go (hunk 2 above)
Comment (daxmc99) [reply drivers/vmwarevsphere/cloudinit.go]: https://github.com/canonical/cloud-init/blob/master/cloudinit/config/cc_users_groups.py#L28-L71

Comment (luthermonson) [reply drivers/vmwarevsphere/cloudinit.go]: yup, added.

**Pull request merged**